packages/components/src/components/Menu/Menu.tsx

```diff
@@ -40,24 +45,42 @@ const FullWidthButton = ({ ...props }) => (
 
 /** A menu is a button element that opens a menu with items. */
 const Menu = React.forwardRef<HTMLButtonElement, BoxProps & MenuProps>(
-  ({ menuButton, items, menuZIndex = "auto", ...props }, ref) => {
-    const store = useMenuStore();
+  (
+    {
+      menuButton,
+      items,
+      menuZIndex = "auto",
+      customStore,
+      menuButtonProps,
+      ...props
+    },
+    ref,
+  ) => {
+    const store = useMenuStore({
+      store: customStore,
+    });
 
     const button = menuButton || VerticalEllipsisButton;
 
     return (
-      <Box.div {...props}>
+      <>
```

Comment:
When removing this, we will need to fix the zIndex in the classic app, for example, because the prop won't be accepted there anymore. That's why I created the breaking change.